File: python/appconstants.py
import os

CURRENT_SEASON = '2122'
LAST_SEASON = '2021'

absolutepath = os.path.abspath(__file__)

fileDir = os.path.dirname(absolutepath)

parentDir = os.path.dirname(fileDir)

mainFinished = os.path.join(fileDir, 'mainFinished.json')

signInFinished = os.path.join(fileDir, 'signInFinished.json')

storedDataPath = os.path.join(fileDir, 'stored_data')

playersHistory = os.path.join(fileDir, 'players_history.json')
historyUrl = 'https://fantasy.premierleague.com/api/element-summary/{element_id}/'

generalInfoUrl = 'https://fantasy.premierleague.com/api/bootstrap-static/'
fixturesUrl = 'https://fantasy.premierleague.com/api/fixtures/'
eventUrl = 'https://fantasy.premierleague.com/api/event/{event_id}/live/'

generalInfoJsonDir = os.path.join(storedDataPath, 'general-info.json')
fixturesJsonDir = os.path.join(storedDataPath, 'fixtures.json')

teamsPath = os.path.join(fileDir, 'teams')

bvsjsonDir = os.path.join(teamsPath, 'best_value_season_team.json')

tablesPath = os.path.join(fileDir, 'tables')

playersTableDir = os.path.join(tablesPath, 'playersTable.json')
leagueTableDir = os.path.join(tablesPath, 'leagueTable.json')
resultsTableDir = os.path.join(tablesPath, 'resultsTable.json')
fixtureTableDir = os.path.join(tablesPath, 'fixtureTable.json')

badgesPath = os.path.join(fileDir, 'badges')

player_images_link_1 = 'https://resources.premierleague.com/premierleague/photos/players/110x140/p'
# element code
player_images_link_2 = '.png'

signInDataPath = os.path.join(storedDataPath, 'signInData')
predictionsPath = os.path.join(storedDataPath, 'predictions')

managerSecurityDir = os.path.join(signInDataPath, 'manager-security.json')

managerBasicInfoJsonDir = os.path.join(signInDataPath, 'manager-basic-info.json')
predBasicInfoJsonDir = os.path.join(predictionsPath, 'manager-basic-info.json')

managerListOfGameweeksJsonDir = os.path.join(signInDataPath, 'list-of-gameweeks.json')
predListOfGameweeksJsonDir = os.path.join(predictionsPath, 'list-of-gameweeks.json')

managerHistoryPath = os.path.join(signInDataPath, 'history')
managerHistoryJsonDir = os.path.join(managerHistoryPath, 'gameweek-{event_id}.json')

predHistoryPath = os.path.join(predictionsPath, 'history')
predHistoryJsonDir = os.path.join(predHistoryPath, 'gameweek-{event_id}.json')

managerTeamsPath = os.path.join(signInDataPath, 'teams')
managerTeamPerGameweekJsonDir = os.path.join(managerTeamsPath, 'gameweek-{event_id}.json')

predTeamsPath = os.path.join(predictionsPath, 'teams')
predTeamPerGameweekJsonDir = os.path.join(predTeamsPath, 'gameweek-{event_id}.json')

managerBasicInfoUrl = 'https://fantasy.premierleague.com/api/entry/{manager_id}/'
managerHistoryUrl = 'https://fantasy.premierleague.com/api/entry/{manager_id}/history/'
managerTeamPerGameweekUrl = 'https://fantasy.premierleague.com/api/entry/{manager_id}/event/{event_id}/picks/'

fpl_data_path = fileDir + '/fpldata/'

players_1617 = fpl_data_path + '16-17/players.csv'
players_1718 = fpl_data_path + '17-18/players.csv'
players_1819 = fpl_data_path + '18-19/players.csv'
players_1920 = fpl_data_path + '19-20/players.csv'
players_2021 = fpl_data_path + '20-21/players.csv'
players_2122 = fpl_data_path + '21-22/players.csv'

gameweeks_1617 = fpl_data_path + '16-17/gameweeks.csv'
gameweeks_1718 = fpl_data_path + '17-18/gameweeks.csv'
gameweeks_1819 = fpl_data_path + '18-19/gameweeks.csv'
gameweeks_1920 = fpl_data_path + '19-20/gameweeks.csv'
gameweeks_2021 = fpl_data_path + '20-21/gameweeks.csv'
gameweeks_2122 = fpl_data_path + '21-22/gameweeks.csv'

team_codes = fpl_data_path + 'teams.csv'

File: src/src/main/java/AppConstants.java
import java.io.File;

public class AppConstants {
    public static class directories {
        public static final String CWD_PARENT_DIR = new File(System.getProperty("user.dir")).getParent(); //>-1
        public static final String IMAGES_DIR = CWD_PARENT_DIR + "/images"; //>-0
        public static final String PYTHON_DIR = CWD_PARENT_DIR + "/python"; //>-0
        public static final String STORED_DATA_DIR = PYTHON_DIR + "/stored_data"; //>1
        public static final String TABLES_DIR = PYTHON_DIR + "/tables"; //>-2

        public static final String PREDICTIONS_DIR = STORED_DATA_DIR + "/predictions"; //>2
        public static final String PREDICTIONS_HISTORY_DIR = PREDICTIONS_DIR + folders.HISTORY_FOLDER; //>3
        public static final String PREDICTIONS_TEAMS_DIR = PREDICTIONS_DIR + folders.TEAMS_FOLDER; //>3

        public static final String SIGN_IN_DIR = STORED_DATA_DIR + "/signInData"; //>2
        public static final String SIGN_IN_HISTORY_DIR = SIGN_IN_DIR + folders.HISTORY_FOLDER; //>3
        public static final String SIGN_IN_TEAMS_DIR = SIGN_IN_DIR + folders.TEAMS_FOLDER; //>3
    }
    public static class folders {
        public static final String HISTORY_FOLDER = "/history";
        public static final String TEAMS_FOLDER = "/teams";
    }

    public static class fileNames {
        public static final String GAMEWEEK_FILE_NAME = "/gameweek-{gameweek}.json";
        public static final String LOGW_FILE_NAME = "/list-of-gameweeks.json";
        public static final String MANAGER_BASIC_FILE_NAME = "/manager-basic-info.json";
        public static final String MANAGER_SECURITY_FILE_NAME = "/manager-security.json";
    }
    public static class paths {
        public static final String PYTHON_MAIN_PATH = directories.PYTHON_DIR + "/main.py"; //>1
        public static final String PYTHON_MAIN_FINISHED_PATH = directories.PYTHON_DIR + "/mainFinished.json"; //>1
        public static final String SIGN_IN_PATH = directories.PYTHON_DIR + "/signIn.py"; //>1
        public static final String SIGN_IN_FINISHED_PATH = directories.PYTHON_DIR + "/signInFinished.json"; //>1

        public static final String PREDICTIONS_LOGW_PATH = directories.PREDICTIONS_DIR + fileNames.LOGW_FILE_NAME; //>3
        public static final String PRED_MANAGER_BASICS_PATH = directories.PREDICTIONS_DIR + fileNames.MANAGER_BASIC_FILE_NAME; //>3
        public static final String PREDICTIONS_HISTORY_PATH = directories.PREDICTIONS_HISTORY_DIR + fileNames.GAMEWEEK_FILE_NAME; //>4
        public static final String PREDICTIONS_TEAMS_PATH = directories.PREDICTIONS_TEAMS_DIR + fileNames.GAMEWEEK_FILE_NAME; //>4

        public static final String SIGN_IN_LOGW_PATH = directories.SIGN_IN_DIR + fileNames.LOGW_FILE_NAME; //>3
        public static final String SIGN_IN_MANAGER_BASICS_PATH = directories.SIGN_IN_DIR + fileNames.MANAGER_BASIC_FILE_NAME; //>3
        public static final String SIGN_IN_MANAGER_SECURITY_PATH = directories.SIGN_IN_DIR + fileNames.MANAGER_SECURITY_FILE_NAME; //>3
        public static final String SIGN_IN_HISTORY_PATH = directories.SIGN_IN_HISTORY_DIR + fileNames.GAMEWEEK_FILE_NAME; //>4
        public static final String SIGN_IN_TEAMS_PATH = directories.SIGN_IN_TEAMS_DIR + fileNames.GAMEWEEK_FILE_NAME; //>4

        public static final String PLAYERS_TABLE_PATH = directories.TABLES_DIR + "/playersTable.json"; //>3
        public static final String LEAGUE_TABLE_PATH = directories.TABLES_DIR + "/leagueTable.json"; //>3
        public static final String FIXTURE_TABLE_PATH = directories.TABLES_DIR + "/fixtureTable.json"; //>3
        public static final String RESULTS_TABLE_PATH = directories.TABLES_DIR + "/resultsTable.json"; //>3

        public static final String PL_LOGO_GREEN = directories.IMAGES_DIR + "/pl_logo_green.png";
        public static final String PL_LOGO_PURPLE = directories.IMAGES_DIR + "/pl_logo_purple.png";
    }
    public static class urls {
        public static final String PLAYER_IMAGE_URL = "https://resources.premierleague.com/premierleague/photos/players/110x140/p";
    }
    public static class program {
        public static final String PYTHON_TYPE = "Python3";
    }
}